File: JavaStackQueue/edu/duq/mathcs/COSC435/Queue.java
/**
 * Represents a generic queue.
 * Implemented using the SingleLinkedList generic class.
 */
package edu.duq.mathcs.COSC435; 

public class Queue<T> 
{
    /**
     * A queue is represented using a SingleLinkedList.
     */
    private SingleLinkedList<T> queue = new SingleLinkedList<T>();

    // IMPLEMENT BELOW GENERIC PUBLIC METHODS WITH THE FOLLOWING NAMES
    //   enqueue
	public void enqueue(T en) {
		queue.insertAtTail(en); 
	}
    //   dequeue
	public T dequeue() {
		return queue.removeAtHead(); 
	}
}
    
    

File: C++StackQueue/queue.cpp
#include "queue.h"

  void queue::enqueue(int value) {
    // Simulate enqueue operation (add to queue) using this object's
    // linked list.
	q.insert_at_tail(value); 
  }
  int queue::dequeue() {
    // Simulate dequeue operation (remove from queue) using this
    // object's linked list.
	return q.remove_at_head(); 
  }

